util/netutil.py

```diff
@@ -113,6 +112,51 @@ def isdir(path, sftp):
     except IOError:
         return False
 
+def remote_file_exists(path, filename, ssh):
+    """Checks if a given remote file exists
+    :param path: A string with the full path where the file to be checked has be placed
+    :param filename: A file under the param path we want to check
+    :param ssh: An sftp connection object (paramiko)
+    :returns: True if the given path is a directory false otherwise.
+    :rtype: bool
+    :type path: str
+    :type filename: str 
+    :type ssh: paramiko.SFTPClient
+    """
+
+    try:
+        sftp = ssh.open_sftp()
+        sftp.chdir(path)
+        sftp.stat(filename)
+
+    except IOError:
+        return False
+
+def check_remote_file(path_file, ssh, pattern):
```

Comment:
I would suggest omitting this function entirely, since it would add maintenace and unit testing overhead.

It could be replaced with something like `ssh_remote_command(...cmd...)`, where `cmd = 'grep <pattern> <file>'`. Then you only need to check whether an empty string or not is being returned.